src/KeyVault/KeyVault/Models/PSKeyVaultCertificatePolicy.cs

```diff
@@ -274,6 +281,147 @@ internal CertificatePolicy ToCertificatePolicy()
+                    SubjectAlternativeNames.Emails.Add(email);
+            if (UserPrincipalNames != null && UserPrincipalNames.Any())
+                foreach (var principalName in UserPrincipalNames)
+                    SubjectAlternativeNames.UserPrincipalNames.Add(principalName);
+            // SubjectAlternativeNames.UserPrincipalNames.Add(principalName);
+            return SubjectAlternativeNames;
+        }
+
+        internal Track2Certificate.CertificatePolicy ToTrack2CertificatePolicy()
+        {
+            Track2Certificate.CertificatePolicy certificatePolicy;
+            if (!string.IsNullOrWhiteSpace(IssuerName) || !string.IsNullOrWhiteSpace(SubjectName) || HasSubjectAlternativeNames())
+            {
+                if (!string.IsNullOrWhiteSpace(SubjectName) && HasSubjectAlternativeNames())
+                {
+                    Track2Certificate.SubjectAlternativeNames subjectAlternativeNames = new Track2Certificate.SubjectAlternativeNames();
+                    subjectAlternativeNames = SetSubjectAlternativeNames(subjectAlternativeNames);
+                    certificatePolicy = new Track2Certificate.CertificatePolicy(IssuerName, SubjectName, subjectAlternativeNames);
+                    
+                }
+                else if (!string.IsNullOrWhiteSpace(SubjectName))
+                {
+                    certificatePolicy = new Track2Certificate.CertificatePolicy(IssuerName, SubjectName);
+                }
+                else if (HasSubjectAlternativeNames())
+                {
+                    Track2Certificate.SubjectAlternativeNames subjectAlternativeNames = new Track2Certificate.SubjectAlternativeNames();
+                    subjectAlternativeNames = SetSubjectAlternativeNames(subjectAlternativeNames);
+                    certificatePolicy = new Track2Certificate.CertificatePolicy(IssuerName, subjectAlternativeNames);
+                }
+                else
+                    certificatePolicy = new Track2Certificate.CertificatePolicy(IssuerName, SubjectName);
+            }else
+            {
+                certificatePolicy = new Track2Certificate.CertificatePolicy();
+            }
+            if (!string.IsNullOrWhiteSpace(SecretContentType))
+                certificatePolicy.ContentType = SecretContentType;
+
+            if (!string.IsNullOrWhiteSpace(Kty))
+                certificatePolicy.KeyType = Kty;
+
+            if (KeySize.HasValue)
+                certificatePolicy.KeySize = KeySize;
+
+            if (!string.IsNullOrWhiteSpace(Curve))
+                certificatePolicy.KeyCurveName = Curve;
+
+            if (ReuseKeyOnRenewal.HasValue)
+                certificatePolicy.ReuseKey = ReuseKeyOnRenewal;
+
+            if (Exportable.HasValue)
+                certificatePolicy.Exportable = Exportable;
+
+            if (CertificateTransparency.HasValue)
+                certificatePolicy.CertificateTransparency = CertificateTransparency;
```

Comment:
```suggestion
            if (CertificateTransparency.HasValue)
                certificatePolicy.CertificateTransparency = CertificateTransparency.Value;
```

Reply:
Maybe an unnecessary change because `certificatePolicy.CertificateTransparency` is `bool?` as well.